freud/order.pyx

```diff
@@ -1347,3 +1347,102 @@ cdef class SolLiqNear(SolLiq):
             self.m_box, points, points, self.num_neigh, nlist, True, self.rmax)
         cdef freud.locality.NeighborList nlist_ = defaulted_nlist[0]
         return SolLiq.computeSolLiqNoNorm(self, points, nlist_)
+
+
+cdef class RotationalAutocorrelation:
+    """Calculates a measure of total rotational autocorrelation based on
+    hyperspherical harmonics as laid out in "Design rules for engineering
+    colloidal plastic crystals of hard polyhedra – phase behavior and
+    directional entropic forces" by Karas et al. (currently in preparation).
+    The output is not a correlation function, but rather a scalar value that
+    measures total system orientational correlation with an initial state. As
+    such, the output can be treated as an order parameter measuring degrees of
+    rotational (de)correlation. For analysis of a trajectory, the compute call
+    needs to be done at each trajectory frame.
+
+    .. moduleauthor:: Andrew Karas
+    .. moduleauthor:: Vyas Ramasubramani
+
+    Args:
+        l (int):
+            Order of the hyperspherical harmonic. Must be a positive, even
+            integer.
+
+    Attributes:
+        num_orientations (unsigned int):
+            The number of orientations used in computing the last set.
+        azimuthal (int):
+            The azimuthal quantum number, which defines the order of the
+            hyperspherical harmonic. Must be a positive, even integer.
+        ra_array (:class:`numpy.ndarray`):
```

Comment:
I don't like the name but `rotational_autocorrelation` is so long... Also do we need to provide a shape for the array?